packages/flutter/lib/src/rendering/object.dart

```diff
@@ -453,6 +453,30 @@ class PaintingContext extends ClipContext {
     return PaintingContext(childLayer, bounds);
   }
 
+  void _debugCheckForClipBehavior(String type) {
+    assert(() {
+      final RenderObject active = RenderObject.debugActivePaint!;
+      active.visitChildren((RenderObject child) {
+        final Rect? approximatePaintClip = active.describeApproximatePaintClip(child);
```

Comment:
What if I have two children, but I only want to clip one in my current config and to do that I call `pushClipRect(clipBehavior: Clip.none)` for one child and `pushClipRect(clipBehavior: Clip.hardEdge)` for the other one. `describeApproximatePaintClip` would return null for one of the children, but non-null for the other.

In that case, wouldn't I incorrectly see this error?

Reply:
(I didn't see a test proving that this would work)

Reply:
I think this error should work in that case, but I will add a test.

This seems like a very strange thing to do though, and it's not how any of the framework ROs are implemented today.

Reply:
Added another test for this in object_test.dart.

Reply:
I don't really understand how this can work for that case... the _debugCheckForClipBehavior method checks all the children, but is called for each pushed clip layer, which might only involve one child.

In general I don't think the PaintingContext has enough information to make this assert work, since it doesn't know which children are being painted in which layer. Some children might not be painted at all, some might be painted in one layer, some another, etc.